rules-java/api/src/main/java/org/jboss/windup/rules/apps/java/scan/provider/DiscoverDuplicateArchiveProjectsRuleProvider.java

```diff
@@ -0,0 +1,68 @@
+package org.jboss.windup.rules.apps.java.scan.provider;
+
+import org.apache.commons.lang.StringUtils;
+import org.jboss.windup.config.AbstractRuleProvider;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.metadata.RuleMetadata;
+import org.jboss.windup.config.operation.GraphOperation;
+import org.jboss.windup.config.operation.iteration.AbstractIterationOperation;
+import org.jboss.windup.config.phase.DiscoverProjectStructurePhase;
+import org.jboss.windup.config.query.Query;
+import org.jboss.windup.graph.GraphContext;
+import org.jboss.windup.graph.model.DuplicateArchiveModel;
+import org.jboss.windup.graph.model.DuplicateProjectModel;
+import org.jboss.windup.graph.model.ProjectModel;
+import org.jboss.windup.graph.model.WindupConfigurationModel;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.graph.service.GraphService;
+import org.jboss.windup.graph.service.ProjectService;
+import org.jboss.windup.graph.service.WindupConfigurationService;
+import org.jboss.windup.rules.apps.java.model.project.MavenProjectModel;
+import org.ocpsoft.rewrite.config.Configuration;
+import org.ocpsoft.rewrite.config.ConfigurationBuilder;
+import org.ocpsoft.rewrite.context.EvaluationContext;
+
+/**
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
```

Comment:
Creates the DuplicateProjectModel's for the DuplicateArchiveModel's and links to the canonical projects.